File: typescript/src/model/definition.ts
export class Definition {
    constructor(
        public definitionText: string | null = null,
        public definitionUrl: string | null = null
    ) { }

    toMap(): Record<string, any> {
        return {
            'definitionText': this.definitionText,
            'definitionUrl': this.definitionUrl,
        };
    }

    static fromMap(data: Record<string, any>): Definition {
        const definitionText = data['definitionText'] ?? null;
        const definitionUrl = data['definitionUrl'] ?? null;

        return new Definition(definitionText, definitionUrl);
    }
}

File: dart/lib/src/model/definition.dart
import 'package:either_dart/either.dart';
import 'package:openral_core/src/cross/backend/parsing/parsing_error.dart';
import 'package:openral_core/src/model_parser/definition_parser.dart';

class Definition {
  String? definitionText;
  String? definitionURL;

  Definition({
    this.definitionText,
    this.definitionURL,
  });

  Map<String, dynamic> toMap() {
    return {
      'definitionText': definitionText,
      'definitionURL': definitionURL,
    };
  }

  static Either<Definition, ParsingError> fromMap(map) {
    final definitionParser = DefinitionParser();

    return definitionParser.fromMap(map);
  }
}
